cluster/saltbase/salt/kube-addons/kube-addons.sh

```diff
@@ -19,6 +19,39 @@
 # managed result is of that. Start everything below that directory.
 KUBECTL=/usr/local/bin/kubectl
 
+function create-kubernetesauth-secret() {
+  token=$1
+  username=$2
+  safe_username=$(tr -s ':_' '--' <<< "${username}")
+
+  # Make secret with a kubernetes_auth file with a token.
+  # TODO(etune): put apiserver certs into secret too, and reference from authfile,
+  # so that "Insecure" is not needed.
+  kafile=$(echo "{\"BearerToken\": \"${token}\", \"Insecure\": true }" | base64 -w0)
+  read -r -d '' secretjson <<EOF
+apiVersion: v1beta1
+kind: Secret 
+id: token-${safe_username}
+data:
+  kubernetes-auth: $kafile
+EOF
+  create-immediate-object "$secretjson"
+# TODO: label the secrets with special label so kubectl does not show these?
+}
+   
+function create-immediate-object() {
+  obj="$1"
+
+  for tries in {1..5}; do
+    if ( echo "${obj}" | ${KUBECTL} --server="127.0.0.1:8080" create -f - ); then
```

Comment:
ok.  there was a bug in kubectl with validate and some objects, but it is since fixed.